Build failure with vala 0.50.1.214-40a6

Vala-Language-Server is not buildable anymore with the recent vala compiler.
```
vala --version
Vala 0.50.1.214-40a6
```
meson configure without a problem but ninja emits those errors

```
../src/list_symbols.vala:387.5-387.35: error: `Vls.ListSymbols.visit_loop': no suitable method found to override
    public override void visit_loop (Vala.Loop stmt) {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../src/codehelp/find_symbol.vala:502.5-502.35: error: `Vls.FindSymbol.visit_loop': no suitable method found to override
    public override void visit_loop (Vala.Loop stmt) {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../src/codehelp/find_scope.vala:204.5-204.35: error: `Vls.FindScope.visit_loop': no suitable method found to override
    public override void visit_loop (Vala.Loop stmt) {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../src/codehelp/symbolvisitor.vala:454.5-454.35: error: `Vls.SymbolVisitor.visit_loop': no suitable method found to override
    public override void visit_loop (Loop stmt) {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 4 error(s), 0 warning(s)
ninja: build stopped: subcommand failed.
```

--- a/src/codehelp/find_scope.vala
+++ b/src/codehelp/find_scope.vala
@@ -201,7 +201,7 @@ class Vls.FindScope : Vala.CodeVisitor {
         local.accept_children (this);
     }
 
-    public override void visit_loop (Vala.Loop stmt) {
+    public override void visit_loop_statement (Vala.LoopStatement stmt) {
         stmt.accept_children (this);
     }
 

--- a/src/codehelp/find_scope.vala
+++ b/src/codehelp/find_scope.vala
@@ -200,8 +200,11 @@ class Vls.FindScope : Vala.CodeVisitor {
     public override void visit_local_variable (Vala.LocalVariable local) {
         local.accept_children (this);
     }
-
+    #if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
+    #else
+    public override void visit_loop (Vala.Loop stmt) {
+    #endif
         stmt.accept_children (this);
     }
 

--- a/src/codehelp/find_scope.vala
+++ b/src/codehelp/find_scope.vala
@@ -200,11 +200,11 @@ class Vls.FindScope : Vala.CodeVisitor {
     public override void visit_local_variable (Vala.LocalVariable local) {
         local.accept_children (this);
     }
-    #if VALA_0_52
+#if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
-    #else
+#else
     public override void visit_loop (Vala.Loop stmt) {
-    #endif
+#endif
         stmt.accept_children (this);
     }
 

--- a/src/codehelp/find_symbol.vala
+++ b/src/codehelp/find_symbol.vala
@@ -499,7 +499,11 @@ class Vls.FindSymbol : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
+    #if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
+    #else
+    public override void visit_loop (Vala.Loop stmt) {
+    #endif
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/codehelp/find_symbol.vala
+++ b/src/codehelp/find_symbol.vala
@@ -499,7 +499,7 @@ class Vls.FindSymbol : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
-    public override void visit_loop (Vala.Loop stmt) {
+    public override void visit_loop_statement (Vala.LoopStatement stmt) {
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/codehelp/find_symbol.vala
+++ b/src/codehelp/find_symbol.vala
@@ -499,11 +499,11 @@ class Vls.FindSymbol : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
-    #if VALA_0_52
+#if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
-    #else
+#else
     public override void visit_loop (Vala.Loop stmt) {
-    #endif
+#endif
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/codehelp/symbolvisitor.vala
+++ b/src/codehelp/symbolvisitor.vala
@@ -451,7 +451,7 @@ class Vls.SymbolVisitor<G> : CodeVisitor {
         stmt.accept_children (this);
     }
 
-    public override void visit_loop (Loop stmt) {
+    public override void visit_loop_statement (LoopStatement stmt) {
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/codehelp/symbolvisitor.vala
+++ b/src/codehelp/symbolvisitor.vala
@@ -451,11 +451,11 @@ class Vls.SymbolVisitor<G> : CodeVisitor {
         stmt.accept_children (this);
     }
 
-    #if VALA_0_52
+#if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
-    #else
+#else
     public override void visit_loop (Vala.Loop stmt) {
-    #endif
+#endif
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/codehelp/symbolvisitor.vala
+++ b/src/codehelp/symbolvisitor.vala
@@ -451,7 +451,11 @@ class Vls.SymbolVisitor<G> : CodeVisitor {
         stmt.accept_children (this);
     }
 
-    public override void visit_loop_statement (LoopStatement stmt) {
+    #if VALA_0_52
+    public override void visit_loop_statement (Vala.LoopStatement stmt) {
+    #else
+    public override void visit_loop (Vala.Loop stmt) {
+    #endif
         if (seen.contains (stmt))
             return;
         seen.add (stmt);

--- a/src/list_symbols.vala
+++ b/src/list_symbols.vala
@@ -384,7 +384,11 @@ class Vls.ListSymbols : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
+    #if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
+    #else
+    public override void visit_loop (Vala.Loop stmt) {
+    #endif
         if (stmt.source_reference != null && stmt.source_reference.file != file) return;
         stmt.accept_children (this);
     }

--- a/src/list_symbols.vala
+++ b/src/list_symbols.vala
@@ -384,11 +384,11 @@ class Vls.ListSymbols : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
-    #if VALA_0_52
+#if VALA_0_52
     public override void visit_loop_statement (Vala.LoopStatement stmt) {
-    #else
+#else
     public override void visit_loop (Vala.Loop stmt) {
-    #endif
+#endif
         if (stmt.source_reference != null && stmt.source_reference.file != file) return;
         stmt.accept_children (this);
     }

--- a/src/list_symbols.vala
+++ b/src/list_symbols.vala
@@ -384,7 +384,7 @@ class Vls.ListSymbols : Vala.CodeVisitor {
         stmt.accept_children (this);
     }
 
-    public override void visit_loop (Vala.Loop stmt) {
+    public override void visit_loop_statement (Vala.LoopStatement stmt) {
         if (stmt.source_reference != null && stmt.source_reference.file != file) return;
         stmt.accept_children (this);
     }

